src/main/scala/mesosphere/marathon/scheduling/Scheduler.scala

```diff
@@ -0,0 +1,76 @@
+package mesosphere.marathon
+package scheduling
+
+import akka.Done
+import mesosphere.marathon.core.instance.Instance
+import mesosphere.marathon.core.launcher.OfferProcessor
+import mesosphere.marathon.state.PathId
+import org.apache.mesos.Protos
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait Scheduler extends OfferProcessor {
+
+  //def create(runSpec: RunSpec): Future[Instance.Id]
```

Comment:
Why is creating an Instance not part of the interface?

Reply:
It should but
* I did not really know what it should look like.
* With the lanuch queue still in place it's not trivial to actually implement the interface.

Reply:
To elaborate, `LaunchQueue.Add` does three things
* It looks for stopped instances and sets their goal to running again.
* It create instances if there are not enough stopped instances.
* It start the `TaskLauncherActor` for the runs spec.

That behaviour is not inline with the changes in https://github.com/mesosphere/marathon/pull/6340, ie phase 2. Phase 2 aims to enable a CRUD like interface. Instead of an add we want a `def create(runSpec)` and `updateGoal(runSpec, goal)`. Phase 3 is supposed to enable that by removing the add from the launch queue. However, that requires a refactoring of the `TaskLauncherActor` to a pure function or rather move the actor into `OfferMatcherManagerActor`.